tests/services/repo-creator.service.spec.ts

```diff
@@ -37,6 +41,32 @@ describe('RepoCreatorService', () => {
     });
   });
 
+  describe('.requestRepoCreationPermissions()', () => {
+    it('should not return any permissions if the repo was already created', () => {
+      of(true)
+        .pipe(repoCreatorService.requestRepoCreationPermissions(Phase.phaseBugReporting, PHASE_REPO))
+        .subscribe((repoCreationPermission: boolean | null) => expect(repoCreationPermission).toBe(null));
+    });
+
+    it('should not return any permissions if the repo was not created and the current phase is not bugReporting', () => {
+      of(false)
+        .pipe(repoCreatorService.requestRepoCreationPermissions(Phase.phaseModeration, PHASE_REPO))
+        .subscribe((repoCreationPermission: boolean | null) => expect(repoCreationPermission).toBe(null));
+    });
+
+    it('should return permissions from matDialog if the repo was not created and current phase is bugReporting', () => {
+      userService.currentUser = USER_JUNWEI;
+      matDialog.open.and.callFake(() => matDialogRef);
+      matDialogRef.afterClosed.and.callFake(() => of(false));
+      of(false)
+        .pipe(repoCreatorService.requestRepoCreationPermissions(Phase.phaseBugReporting, PHASE_REPO))
+        .subscribe((repoCreationPermission: boolean | null) => expect(repoCreationPermission).toBe(false));
```

Comment:
```suggestion
        .subscribe((repoCreationPermission: boolean | null) => expect(repoCreationPermission).toBe(permissionFromDialog));
```

Will be better to create a variable `permissionFromDialog` and use it in the `callFake` callback, as well as in the expected result here.

This makes it clearer that we expect the permission to match the value provided by the dialog.

Reply:
Ah alright, this sounds better to do! :+1: